packages/pigeon/example/app/lib/src/messages.g.dart

```diff
@@ -29,6 +29,23 @@ List<Object?> wrapResponse(
   return <Object?>[error.code, error.message, error.details];
 }
 
+bool _deepEquals(Object? a, Object? b) {
+  if (a is List && b is List || a is Map && b is Map) {
+    if (a is List && b is List) {
+      a = a.asMap();
+      b = b.asMap();
+    }
+    final Map<Object?, Object?> a1 = a! as Map<Object?, Object?>;
+    final Map<Object?, Object?> b1 = b! as Map<Object?, Object?>;
+
+    final List<Object?> keys = a1.keys.toList();
```

Comment:
The `toList` here is also constructing an entirely new data structure, when all you need is the `Iterable` that `keys` returns.